nrf_security/Kconfig

```diff
@@ -0,0 +1,477 @@
+	default y
+	depends on CC310_BACKEND || VANILLA_BACKEND
+	help
+	  Enable the ECDH module.
+	  MBEDTLS_ECDH_C setting in mbedtls config file.
+
+endif # !NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+if NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+menu "ECDH  - Elliptic Curve Diffie-Hellman library support"
+
+config GLUE_MBEDTLS_ECDH_C
+	bool
+
+config CC310_MBEDTLS_ECDH_C
+	bool "cc310" if CC310_BACKEND
+	default y
+	help
+	  Enable the ECDH library in nrf cc310.
+	  MBEDTLS_ECDH_C setting in mbedtls config file.
+
+config VANILLA_MBEDTLS_ECDH_C
+	bool "mbedtls vanilla" if VANILLA_BACKEND
+	select GLUE_MBEDTLS_ECDH_C if CC310_MBEDTLS_ECDH_C
+	help
+	  Enable the ECDH module from mbedtls vanilla.
+	  MBEDTLS_ECDH_C setting in mbedtls config file.
+endmenu
+
+endif # NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+if !NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+config MBEDTLS_ECDSA_C
+	bool
+	prompt "ECDSA - Elliptic Curve Digital Signature Algorithm support"
+	default y
+	depends on CC310_BACKEND || VANILLA_BACKEND
+	help
+	  Enable the ECDSA module.
+	  MBEDTLS_ECDSA_C setting in mbedtls config file.
+
+endif # !NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+if NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+menu "ECDSA - Elliptic Curve Digital Signature Algorithm support"
+
+config GLUE_MBEDTLS_ECDSA_C
+	bool
+
+config CC310_MBEDTLS_ECDSA_C
+	bool "cc310" if CC310_BACKEND
+	default y
+	help
+	  Enable the ECDSA library in nrf cc310.
+	  MBEDTLS_ECDSA_C setting in mbedtls config file.
+
+config VANILLA_MBEDTLS_ECDSA_C
+	bool "mbedtls vanilla" if VANILLA_BACKEND
+	select GLUE_MBEDTLS_ECDSA_C if CC310_MBEDTLS_ECDSA_C
+	help
+	  Enable the ECDSA module from mbedtls vanilla.
+	  MBEDTLS_ECDSA_C setting in mbedtls config file.
+endmenu
+
+endif # NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+
+config MBEDTLS_RSA_C
+	bool
+	prompt "RSA   - Rivest–Shamir–Adleman cryptosystem"
+	depends on (CC310_BACKEND || VANILLA_BACKEND)
+	default y
+	help
+	  Enable RSA cryptosystem support.
+	  MBEDTLS_RSA_C setting in mbedtls config file.
+
+choice
+	bool
+	prompt "Backend selection" if MBEDTLS_RSA_C
+	depends on NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+	default CC310_MBEDTLS_RSA_C
+
+	config CC310_MBEDTLS_RSA_C
+	    bool "cc310" if CC310_BACKEND
+
+	config VANILLA_MBEDTLS_RSA_C
+	    bool "mbedtls vanilla" if VANILLA_BACKEND
+endchoice
+
+menuconfig HASH_FUNCTIONS_CONFIGURATION
+	bool
+	prompt "Hash functions configuration"
+	depends on (CC310_BACKEND || VANILLA_BACKEND)
+	default y
+
+if HASH_FUNCTIONS_CONFIGURATION
+
+config MBEDTLS_SHA1_C
+	bool
+	prompt "SHA-1 hash functionality."
+	depends on (CC310_BACKEND || VANILLA_BACKEND)
+	default y
+	help
+	  SHA-1 hash functionality.
+
+choice
+	bool
+	prompt "Backend selection"
+	depends on MBEDTLS_SHA1_C && NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0
+	default CC310_MBEDTLS_SHA1_C
+
+	config CC310_MBEDTLS_SHA1_C
+	    bool "cc310" if CC310_BACKEND
+
+	config VANILLA_MBEDTLS_SHA1_C
+	    bool "mbedtls vanilla" if (VANILLA_BACKEND && NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0)
```

Comment:
Why is `NRF_CRYPTO_MULTIPLE_CHOICE_GROUP_0` checked here?

Reply:
Correct, that's redundant.
will remove